crates/typos-cli/src/file_type.rs

```diff
@@ -107,9 +107,15 @@ impl Types {
     }
 
     pub fn file_matched(&self, path: &std::path::Path) -> Option<&str> {
-        let file_name = path.file_name()?.to_str()?.trim_end_matches(".in");
+        let mut file_name = path.file_name()?.to_str()?;
```

Comment:
Overlooked this before: this will cause us to fail to lookup the type if the file name isn't valid UTF-8 (`.to_str()?`)

Reply:
Maybe e12b9a49a5fa70234376b3224755d192f2ef964a?